framework/actions/aggregate.go

```diff
@@ -21,24 +24,68 @@ import (
 //
 // The rule set passed to each plugin can be configured with compliance specific settings based on the InputContext.
 func AggregateResults(ctx context.Context, inputContext *InputContext, pluginSet map[plugin.ID]policy.Provider) ([]policy.PVPResult, error) {
-	var allResults []policy.PVPResult
 	log := logging.GetLogger("aggregator")
+
+	var allResults []policy.PVPResult
+	sem := semaphore.NewWeighted(inputContext.MaxConcurrentWeight)
+	var wg sync.WaitGroup
+	errorCh := make(chan error, len(pluginSet))
+	resultChan := make(chan policy.PVPResult, len(pluginSet))
+
 	for providerId, policyPlugin := range pluginSet {
-		componentTitle, err := inputContext.ProviderTitle(providerId)
-		if err != nil {
-			return nil, err
-		}
-		log.Debug(fmt.Sprintf("Aggregating results for provider %s", providerId))
-		appliedRuleSet, err := settings.ApplyToComponent(ctx, componentTitle, inputContext.Store(), inputContext.Settings)
-		if err != nil {
-			return allResults, fmt.Errorf("failed to get rule sets for component %s: %w", componentTitle, err)
-		}
-
-		pluginResults, err := policyPlugin.GetResults(appliedRuleSet)
-		if err != nil {
-			return allResults, fmt.Errorf("plugin %s: %w", providerId, err)
-		}
-		allResults = append(allResults, pluginResults)
+
+		wg.Add(1)
+
+		go func(providerId plugin.ID, plugin policy.Provider) {
```

Comment:
I'm wondering if it wouldn't be beneficial to include a timeout in this function in case a plugin stalls.

Reply:
In expected cases we can speed up the assessment, which is great. But I also wonder if we should abort the whole process instead of returning incomplete results. One advantage with this parallelism is that we can abort earlier and save resources unless partial results bring value.